src/libmongoc/doc/mongoc_structured_log_get_component_name.rst

```diff
@@ -0,0 +1,27 @@
+:man_page: mongoc_structured_log_get_component_name
+
+mongoc_structured_log_get_component_name
+========================================
+
+Synopsis
+--------
+
+.. code-block:: c
+
+  const char *
+  mongoc_structured_log_get_component_name (mongoc_structured_log_component_t component);
+
+Parameters
+----------
+
+* ``component``: Log component as a :symbol:`mongoc_structured_log_component_t`.
+
+Returns
+-------
+
+If the component is known, returns a pointer to a constant string that should not be freed.
+If the component has no known name, returns NULL.
```

Comment:
I thought it was better to return NULL than a placeholder string, since different users might want different behavior (like retaining the enum code as part of the string perhaps)